contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -2003,6 +2019,58 @@ tsql_select_common_typmod_hook(ParseState *pstate, List *exprs, Oid common_type)
 	return max_typmods;
 }
 
+/* For CASE expression, this function will calculate typmod from tsql_select_common_typmod_hook(),
+ * and after that we will set this typmod to all the CASE branches from coerce_to_target_type().
+ */
+static Node*
+tsql_set_common_typemod_case_expr_hook(ParseState *pstate, List *exprs, CaseExpr *newc)
+{
+	CaseExpr   *tmpc  =  makeNode(CaseExpr);
+	int32       typmod;
+        ListCell   *l;
+
+        if(sql_dialect != SQL_DIALECT_TSQL)
+                return (Node *) newc;
+
+        tmpc->xpr = newc->xpr;
+        tmpc->casecollid = newc->casecollid;
+        tmpc->location = newc->location;
+        tmpc->arg = newc->arg;
+        tmpc->args = newc->args;
+        tmpc->defresult = newc->defresult;
+        tmpc->casetype = newc->casetype;
```

Comment:
Do we really need new copy? Can't we directly update the previous one?

Reply:
resolved